aten/src/ATen/autocast_mode.h

```diff
@@ -10,6 +10,16 @@
 
 namespace at::autocast {
 
+constexpr std::array<at::DeviceType, 7> kAllAutocastDeviceTypes = {
+    at::kCPU,
+    at::kCUDA,
+    at::kXPU,
+    at::kIPU,
+    at::kXLA,
+    at::kHPU,
+    at::kPrivateUse1,
```

Comment:
similarly, I see a similar list of "autocast devices" in this function: https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/autocast_mode.h#L174

what do you think of either:

(1) figuring out how to expose that same list of devices to python so we have single source of truth

(2) Adding a `Note [Autocast Devices]` comment that we mirror to both call-sites, just so that all of the places in the code that need to change are easier to find if we don't decide to have one source of truth for the device list

Reply:
Yeah I'll do some more cleanup in this PR

Reply:
I'm not going to do the consolidation in this PR, because it has behavior changes